test/Chirp.Infrastructure.Tests/RepositoryTests/AuthorRepositoryTest.cs

```diff
@@ -0,0 +1,261 @@
+using Chirp.Infrastructure.Repository;
+using Chirp.Core.Entities; 
+using Chirp.Infrastructure;
+using Test_Utilities;
+
+namespace Chirp.InfrastructureTests.RepositoryTests;
+
+public class AuthorRepositoryTest
+{
+
+    private readonly ChirpDbContext context;
+
+    public AuthorRepositoryTest()
+    {
+        context = SqliteInMemoryBuilder.GetContext();
+    }
+
+    [Fact]
+    public void GetAuthorByName_ShouldReturnCorrectAuthorDTO()
+    {
+        var authorRepository = new AuthorRepository(context);
+
+        AuthorDTO theAuthor = null;
+        for (int i = 0; i < 34; i++)
+        {
+
+            if (i == 5)
+            {
+                theAuthor = new AuthorDTO
+                {
+                    AuthorId = Guid.NewGuid(),
+                    Name = "TestAuthor" + i,
+                    Email = "mock" + i + "@email.com"
+                };
+                context.Authors.Add(theAuthor);
+                continue;
+            }
+
+            AuthorDTO authorDto = new AuthorDTO
+            {
+                AuthorId = Guid.NewGuid(),
+                Name = "TestAuthor" + i,
+                Email = "mock" + i + "@email.com"
+            };
+
+            CheepDTO cheepDto = new CheepDTO
+            {
+                CheepId = Guid.NewGuid(),
+                AuthorId = authorDto.AuthorId,
+                Text = "TestCheep" + i,
+                AuthorDto = authorDto
+            };
+
+            context.Authors.Add(authorDto);
+            context.Cheeps.Add(cheepDto);
+        }
+
+        context.SaveChanges();
+
+        //Act
+        AuthorDTO expectedAuthor = theAuthor;
+        AuthorDTO returnedAuthor = authorRepository.GetAuthorByName("TestAuthor5");
+
+        //Assert
+        Assert.Equal(expectedAuthor, returnedAuthor);
```

Comment:
Der testes kun for én author. Som et alternativ kan man eventuelt gemme en liste af alle de generede authors og så iterere gennem listen og assert hver author med deres ".GetAuthorByName" modpart

Dette er ikke en nødvendighed, men mere en skalérbar forbedring til testen

Reply:
Det er helt klart ikke noget vi bare kan udelukke at gøre.

Vi skal dog huske på at det er ekstra processing (nok meget lidt, men stadig), som i bund og grund kun beviser at det ikke bare var held første gang.